### DISCUSSION TITLE
GodMode9 v2.1.0

### DISCUSSION THREAD

**Discussion by d0k3** [d0k3/GodMode9#754] [General]
We're early for christmas this year with a new release: GodMode9 v2.1.0 comes with several new features and a truckload of bugfixes and smaller improvements.

This is new since the last release:
* [new] Unicode support, including a special Unicode font created for GodMode9 (@Epicpkmn11)
* [new] Support for mounting `certs.db`, including improvement of certs handling (@luigoalma)
* [new] Support for SHA-1 checksums (@aspargas2)
* [new] Raw cart dumper, accessible via R+A on cart drive
* [improved] Wider support for flash chips on carts, including ir (@dratini0)
* [scripting] `SDSIZE`, `SDFREE` and `NANDFREE` global variables
* [scripting] Added `cartdump` command
* [fixed] Numerous bugfixes and small improvements, too many to list

**Killer Feature: Unicode support**
GodMode9 is no longer limited to just displaying English, it now has proper support for Unicode text. To facilitate this change there's also a new font format, a simple RIFF-based format called FRF, which you can easily convert to using the ´fontriff.py´ conversion script. The default font has been updated to include the common accented Latin letters, Cyrillic for Russian, and Hiragana/Katakana for Japanese and there's also a font with complete JIS X 0208 support in the resources folder. You can also now type any character (up to U+FFFF) in the keyboard by switching to the number input mode, typing the codepoint, and pressing the U+ key. (ex. 0411 is Б and 3042 is あ).

**How do I update GodMode9?**
(Do I really need to add that paragraph everytime?) You wouldn't believe how often we get that question when we do a new release. It's actually very simple: Just replace `GodMode9.firm` on your SD card with the file found in the release ZIP. You may also want to update scripts, which are found in the `./gm9` folder inside the archive and go to the same folder on your SD card.

**Special thanks**
With the project now being over 5 years old, and all of us working on other stuff, additionally to also having a real life(tm), the main devs (@aspargas2, @Wolfvak and me) sometimes take the backseat, only doing smaller features, bug fixes and general project maintenance. Gladly the void was filled by other developers. Special thanks go to @Epicpkmn11 for Unicode support,@dratini0 for his continued work on supporting flash chips in the cart dumper and @luigoalma who added support for mounting `certs.db` in this release. It's not only the devs who we have to thank, though, it's also the numerous people who point out problems and ask for features, as well as everyone who's patient enough to help us out with testing. More special thanks go to @TimmSkiller and @mariomadproductions, who both helped us a lot to improve GodMode9. In fact, the people who helped make this release a possibility are too numerous to list - thanks, all of you!

<hr /><em>This discussion was created from the release <a href='https://github.com/d0k3/GodMode9/releases/tag/v2.1.0'>GodMode9 v2.1.0</a>.</em>

**Comment by FieryMewtwo** [comment]
### There's nothing but source code...

**Comment by arna2020** [comment]
YESSSSSSSSSSSSSS

**Comment by TzakShrike** [comment]
Whoa I'm really happy for the Unicode support! Thank you!
